services/actions/notifier_helper.go

```diff
@@ -232,6 +232,12 @@ func notify(ctx context.Context, input *notifyInput) error {
 			log.Error("jobparser.Parse: %v", err)
 			continue
 		}
+
+		// auto cancel running jobs in the same workflow
+		if err := actions_model.CancelRunningJobs(ctx, run); err != nil {
```

Comment:
So it will cancel the previous jobs of the same ref by default, I'm OK with this, and we can add provide an options later if someone ask for it.

Reply:
@wolfogre Yes, move to another NEW PR.